### PR TITLE
Fix : Resource Leak

### PR DESCRIPTION
**Pull request by FollowMeDown**
ok ?

### CODE REVIEW

**Comment by FollowMeDown** [comment]
Proof of Make :
```
deface@proton ~/_/p/src (master)> make
gcc    -c -o pam2control.o pam2control.c
gcc    -c -o config.o config.c
gcc    -c -o log.o log.c
gcc    -c -o smtp.o smtp.c
gcc    -c -o conv.o conv.c
gcc -Wall -fPIC -c pam2control.c config.c log.c smtp.c conv.c
gcc -shared -o pam2control.so pam2control.o config.o log.o smtp.o conv.o -lpam
```

**Comment by FollowMeDown** [comment]
If va_end is not called before returning from the function, the result is undefined.

**Review by alexander-naumov — CHANGES_REQUESTED** [review]
Well, we use `log_path` in log.c to define PATH of logfile.
You can't just change this name here; `logpath` will be NULL in `history()` func.

**Review by alexander-naumov — APPROVED** [review]
nice

**Review by alexander-naumov — COMMENTED** [review]
ok

**Review by alexander-naumov — COMMENTED** [review]
ok, thanks

File: src/config.c [hunk 1]
```diff
@@ -398,7 +398,7 @@ get_default(settings_t *def)
       def->LOGFILE = strdup(pch);
       rmn(def->LOGFILE);
     }
-    log_path = def->LOGFILE;
+    log_path_conf = def->LOGFILE;
```
Comment (alexander-naumov) [inline src/config.c:401]: Well, we use `log_path` in log.c to define PATH of logfile.
You can't just change this name here; `logpath` will be NULL in `history()` func.

**Review by alexander-naumov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**